python_modules/dagster/dagster/_core/definitions/asset_out.py

```diff
@@ -164,23 +208,45 @@ def to_out(self) -> Out:
     def to_spec(
         self, key: AssetKey, deps: Sequence[AssetDep], additional_tags: Mapping[str, str] = {}
     ) -> AssetSpec:
-        with disable_dagster_warnings():
-            return AssetSpec.dagster_internal_init(
-                key=key,
-                metadata=self.metadata,
-                description=self.description,
-                skippable=not self.is_required,
-                group_name=self.group_name,
-                code_version=self.code_version,
-                freshness_policy=self.freshness_policy,
-                automation_condition=self.automation_condition,
-                owners=self.owners,
-                tags={**additional_tags, **self.tags} if self.tags else additional_tags,
-                deps=deps,
-                auto_materialize_policy=None,
-                partitions_def=None,
-                kinds=None,
-            )
+        return replace_attributes(
+            self._spec,
+            key=key,
+            tags={**additional_tags, **self.tags} if self.tags else additional_tags,
+            deps=deps,
+        )
+
+    @classmethod
+    def from_spec(
+        cls,
```

Comment:
this can just be a static method -- we don't use the cls for anything other than construction and I don't imagine there will be subtypes of AssetOut in the future

Reply:
:nod2: